utils/config_generator.py

```diff
@@ -134,7 +138,8 @@ def update_config(
     else:
         prod_config["continuation"]["models"] = {}
 
-    if is_cjk(source, target):
+    needs_bigger_vocab = {ScriptType.LOGOGRAPHIC, ScriptType.FEATURAL}
+    if src_script["type"] in needs_bigger_vocab or trg_script["type"] in needs_bigger_vocab:
```

Comment:
Those languages need a bigger vocab only when we use a shared vocab. If it's split-vocab we use 32k on each side as before.